src/taco-cli/cli/utils/settings.ts

```diff
@@ -137,78 +137,77 @@ class Settings {
         Settings.settings = null;
     }
 
-    public static determineSpecificPlatformsFromOptions(options: commands.ICommandData, settings: Settings.ISettings): Settings.IPlatformWithLocation[] {
-        var optionsToIgnore = options.original.indexOf("--") === -1 ? [] : options.original.slice(options.original.indexOf("--"));
-        var platforms = options.remain.filter(function (platform: string): boolean { return optionsToIgnore.indexOf(platform) === -1; });
-        // one or more specific platforms are specified. Determine whether they should be built locally, remotely, or local falling back to remote
-        return platforms.map(function (platform: string): Settings.IPlatformWithLocation {
-            var buildLocation: Settings.BuildLocationType;
-            if (options.options["remote"]) {
-                buildLocation = Settings.BuildLocationType.Remote;
-            } else if (options.options["local"]) {
-                buildLocation = Settings.BuildLocationType.Local;
-            } else {
-                // we build remotely if either remote server is setup for the given platform or if the target platform cannot be built locally
-                buildLocation = (platform in (settings.remotePlatforms || {})) || !Settings.canBuildLocally(platform) ?
-                    Settings.BuildLocationType.Remote : Settings.BuildLocationType.Local;
-            }
-
-            return { location: buildLocation, platform: platform };
-        });
-    }
-
     public static loadSettingsOrReturnEmpty(): Q.Promise<Settings.ISettings> {
         return Settings.loadSettings().fail(function (): Settings.ISettings { return { remotePlatforms: {} }; });
     }
 
+    public static parseRequestedPlatforms(options: commands.ICommandData): string[] {
```

Comment:
It does need to be public because some telemetry code needs to call it, to work out what platforms the user asked for.